src/api/artifact/blob/reader.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blob
+
+import (
+	"github.com/docker/distribution/manifest/manifestlist"
+	"github.com/docker/distribution/manifest/schema1"
+	"github.com/docker/distribution/manifest/schema2"
+	coreutils "github.com/goharbor/harbor/src/core/utils"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"io/ioutil"
+)
+
+var (
+	accept = []string{
+		schema1.MediaTypeSignedManifest,
+		schema2.MediaTypeManifest,
+		v1.MediaTypeImageManifest,
+		manifestlist.MediaTypeManifestList,
+		v1.MediaTypeImageIndex,
+	}
+)
+
+// Reader read blob content
+type Reader interface {
+	// Read the content of blob under the repository. Set the "manifest" as true if whether the blob is a manifest
+	Read(repoFullName, digest string, manifest bool) (mediaType string, content []byte, err error)
```

Comment:
Since it does not implement `io.Reader` we should add different funcs like `GetManifest` `GetConfig`...  it's easier for testing.

Reply:
Updated